src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -994,9 +1012,16 @@ static int server_ack_request(sd_dhcp_server *server, DHCPRequest *req, DHCPLeas
         } else {
                 _cleanup_(dhcp_lease_freep) DHCPLease *lease = NULL;
 
-                r = prepare_new_lease(&lease, address, &req->client_id,
-                                      req->message->htype, req->message->hlen,
-                                      req->message->chaddr, req->message->giaddr, expiration);
+                r = prepare_new_lease(
+                                &lease,
+                                address,
+                                &req->client_id,
+                                req->message->htype,
+                                req->message->hlen,
+                                req->message->chaddr,
+                                req->message->giaddr,
+                                expiration,
+                                req->hostname);
```

Comment:
The function already takes many arguments from `req`. Could you make the function take `DHCPRequest*`, rather than adding a new argument?

Reply:
Done.